scripts/stubsabot.py

```diff
@@ -0,0 +1,339 @@
+#!/usr/bin/env python3
+from __future__ import annotations
+
+import argparse
+import asyncio
+import datetime
+import enum
+import functools
+import io
+import os
+import re
+import subprocess
+import sys
+import tarfile
+import urllib.parse
+import zipfile
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any
+
+import aiohttp
+import packaging.specifiers
+import packaging.version
+import tomli
+import tomlkit
+
+
+class ActionLevel(enum.IntEnum):
+    nothing = 0  # make no changes
+    local = 1  # make changes that affect local repo
+    everything = 2  # do everything, e.g. open PRs
```

Comment:
Very minor, but my favourite pattern for things like this is to do something like `http.HTTPStatus`. It allows you to have per-member docstrings:

```suggestion
class ActionLevel(IntEnum):
    def __new__(cls, value: int, doc: str):
        member = int.__new__(cls, value)
        member._value_ = value
        member.__doc__ = doc
        return member

    nothing = 0, "make no changes"
    local = 1, "make changes that affect local repo"
    everything = 2, "do everything, e.g. open PRs"
```